scripts/quests/windurst/Something_Fishy.lua

```diff
@@ -0,0 +1,98 @@
+-----------------------------------
+-- Something Fishy
+-----------------------------------
+-- Log ID: 2, Quest ID: 52
+-- Tokaka : !pos 240 -189.678 -2 66.335
+-----------------------------------
+require('scripts/globals/items')
+require('scripts/globals/npc_util')
+require('scripts/globals/quests')
+require('scripts/globals/zone')
+require('scripts/globals/interaction/quest')
+-----------------------------------
+
+local quest = Quest:new(xi.quest.log_id.WINDURST, xi.quest.id.windurst.SOMETHING_FISHY)
+
+--quest rewards are handled inside the lua based on repeat turn ins.
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE
+        end,
+
+        [xi.zone.PORT_WINDURST] =
+        {
+            ['Tokaka'] = quest:event(208, 0, xi.items.BASTORE_SARDINE),
+
+            onEventFinish =
+            {
+                [208] = function(player, csid, option, npc)
+                    quest:begin(player)
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status ~= QUEST_AVAILABLE
+        end,
+
+        [xi.zone.PORT_WINDURST] =
+        {
+            ['Tokaka'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if
+                        player:getQuestStatus(quest.areaId, quest.questId) ~= QUEST_AVAILABLE and
+                        npcUtil.tradeHasExactly(trade, xi.items.BASTORE_SARDINE) and
+                        quest:getVar(player, 'Dialog') == 1
+                    then
+                        return quest:event(210, 70, xi.items.BASTORE_SARDINE)
+                    end
+                end,
+
+                onTrigger = function(player, npc)
+                    if
+                        player:getQuestStatus(quest.areaId, quest.questId) == QUEST_COMPLETED and
+                        not quest:getMustZone(player) and
+                        quest:getVar(player, 'Dialog') < 1
+                    then
+                        return quest:event(208, 0, xi.items.BASTORE_SARDINE):oncePerZone()
+                    elseif
+                        quest:getVar(player, 'Dialog') == 1 or
+                        player:getQuestStatus(quest.areaId, quest.questId) == QUEST_ACCEPTED
+                    then
+                        return quest:event(209, 0, xi.items.BASTORE_SARDINE)
+                    elseif quest:getMustZone(player) then
+                        return quest:event(211)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [208] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Dialog', 1)
+                end,
+
+                [210] = function(player, csid, option, npc)
```

Comment:
Similar to above, this would probably be cleaner to handle in a separate section for the completed version, and then use the reward table to handle the first time completion.

Reply:
That squash should be a lot cleaner -  tested several times as well and mirrors my capture.